tests/unit/amo/components/TestDownloadFirefoxButton.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import DownloadFirefoxButton, {
+  DownloadFirefoxButtonBase,
+  mapStateToProps,
+} from 'amo/components/DownloadFirefoxButton';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
+import {
+  getFakeI18nInst,
+  shallowUntilTarget,
+  userAgents,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  function render({
+    userAgent = userAgents.firefox[0],
+    ...props
+  } = {}) {
+    const { store } = dispatchClientMetadata({ userAgent });
+
+    return shallowUntilTarget(
+      <DownloadFirefoxButton
+        i18n={getFakeI18nInst()}
+        store={store}
+        {...props}
+      />,
+      DownloadFirefoxButtonBase,
+    );
+  }
+
+  it('renders nothing if the browser is Firefox Desktop', () => {
+    const root = render();
```

Comment:
For tests like this that require a specific property, I think it helps to set it explicitly:

```js
const root = render({ userAgent: userAgents.firefox[0] });
```

Reply:
Agreed, I think I did that earlier then added this test and forgot.